CRM/Utils/Check/Component/Env.php

```diff
@@ -664,24 +667,28 @@ public function checkExtensions() {
 
         default:
           if (in_array($key, $requiredExtensions, TRUE)) {
-            $requiredMessage = new CRM_Utils_Check_Message(
-              __FUNCTION__ . 'Required:' . $key,
-              ts('The extension %1 is required and must be enabled.', [1 => $row['label']]),
-              ts('Required Extension'),
-              \Psr\Log\LogLevel::ERROR,
-              'fa-exclamation-triangle'
-            );
-            $requiredMessage->addAction(
-              ts('Enable %1', [1 => $row['label']]),
-              '',
-              'api3',
-              ['Extension', 'install', ['key' => $key]]
-            );
-            $messages[] = $requiredMessage;
+            $missingRequired[$key] = $row['label'];
           }
       }
     }
 
+    if ($missingRequired) {
+      $requiredMessage = new CRM_Utils_Check_Message(
+        __FUNCTION__ . 'Required:' . implode(',', array_keys($missingRequired)),
+        ts('The extension %1 is required and must be enabled.', [1 => implode(', ', $missingRequired)]),
+        ts('Required Extension'),
+        \Psr\Log\LogLevel::ERROR,
+        'fa-exclamation-triangle'
+      );
+      $requiredMessage->addAction(
+        ts('Enable %1', [1 => implode(', ', $missingRequired)]),
+        '',
+        'api3',
+        ['Extension', 'install', ['keys' => array_keys($missingRequired)]]
```

Comment:
It's a comma-separated list. It's confusing and different depending on the action.

Reply:
An array ought to work
https://github.com/civicrm/civicrm-core/blob/5f23f53c831ca5fe73157c3fcd1610c095830329/api/v3/Extension.php#L409-L411

Reply:
For the api3 `Extension` stuff, it should be agreeable with either an `array` or `string<csv>`: https://github.com/civicrm/civicrm-core/blob/5f23f53c831ca5fe73157c3fcd1610c095830329/api/v3/Extension.php#L409-L415

Reply:
Oh ok. I've had trouble in the past figuring out the params for the Extension.x actions.

Reply:
r-run checks out.
Not a blocker but in the sites I've seen where the upgrade failed you can't get to the status check, but could still be useful.

Reply:
Yeah, I agree it's confusing. 

Historical note: It's because the intended audience included CLI callers (eg `drush civicrm-api extension.install keys=...`). At the time, `cv` didn't exist. If you wanted to pass a clean `array` to the CLI tools, that meant patching 4 projects (cli.php + 2 branches of drush + wp).  It was much more time-effective to call `explode()` than to patch all those projects.

Anyway, yeah, this is the kind of quirk we should eliminate when doing an `Extension` APIv4.